src/Uno.UI/Microsoft/UI/Xaml/Controls/TreeView/TreeViewNodeVector.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Uno.UI;
+using Windows.Foundation.Collections;
+
+namespace Microsoft.UI.Xaml.Controls
+{
+	internal class TreeViewNodeVector : ObservableVector<TreeViewNode>
+	{
+		private TreeViewNode m_parent;
+
+		public TreeViewNodeVector()
+		{
+
+		}
+
+		public void SetParent(TreeViewNode value)
+		{
+			m_parent = value;
+		}
+
+		private IList GetWritableParentItemsSource()
+		{
+			IList parentItemsSource = null;
+
+			if (m_parent?.ItemsSource != null)
+			{
+				parentItemsSource = m_parent?.ItemsSource as IList;
+			}
+
+			return parentItemsSource;
+		}
+
+		public void Append(TreeViewNode item, bool updateItemsSource = true)
+		{
+			InsertAt(Count, item, updateItemsSource);
+		}
+
+		public override void Add(TreeViewNode item) => Append(item);
+
+		public void InsertAt(int index, TreeViewNode item, bool updateItemsSource = true)
+		{
+			var inner = GetVectorInnerImpl();
+			if (m_parent == null)
+			{
+				throw new InvalidOperationException("Parent node must be set");
+			}
+			if (index > inner.Count)
+			{
+				throw new IndexOutOfRangeException("Index out of range for Insert");
+			}
+
+			item.Parent = m_parent;
+
+			inner.Insert(index, item);
+
+			if (updateItemsSource)
+			{
+				var itemsSource = GetWritableParentItemsSource();
+				if (itemsSource != null)
+				{
+					itemsSource.Insert(index, item.Content);
+				}
+			}
+		}
+
+		public override void Insert(int index, TreeViewNode item) => InsertAt(index, item);
+
+		public void SetAt(int index, TreeViewNode item, bool updateItemsSource = true)
+		{
+			RemoveAt(index, updateItemsSource);
+			InsertAt(index, item, updateItemsSource);
+		}
+		
+		public override TreeViewNode this[int index]
+		{
+			get => base[index];
+			set => SetAt(index, value);
+		}
+
+		public void RemoveAt(int index, bool updateItemsSource = true)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This method signature overlaps the one defined on line 103, the default parameter value can't be used.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)